### DISCUSSION TITLE
Improve plugin

### DISCUSSION THREAD

**Discussion by Mister-Hope** [DreamOfIce/vuepress-plugin-i18n#6] [Ideas]
I am extremely interested in this plugin, as this is a common need for all documentation.

I can help to maintain this plugin, but I want to adjust some options to make things clearer. So I need some discussions.

The first thing I want to ask is why are you calling `preparePageComponent` at `extendsPage` lifecyle all the time? At this stage the page is not output into a vue component yet, and an hmr triggers `preparePageComponent` internally. So I think all you need is to edit `Page.sfc` and append the component at the front of the content without any render process or temp file written process.

**Comment by DreamOfIce** [comment]
我也不知道当时怎么想的会加一个`preparePageComponent`，已经改掉了 😂

**Comment by Mister-Hope** [comment]
> sourcePath seems to be the locale path that the original docs locale at, so maybe it should be named something like baseLocalePath?

so any feedback with this one?

**Reply by DreamOfIce** [reply]
Thanks for the suggestion, `sourcePath` is not really accurate. I will think about it tomorrow

**Reply by DreamOfIce** [reply]
Done in 7836e794c6ac89c1ead8d87beef4e9669ec291b0

**Comment by DreamOfIce** [comment]
If there's nothing else, I'll close this tomorrow
